.github/workflows/ci.yaml

```diff
@@ -0,0 +1,203 @@
+name: CI
```

Comment:
It might be better if we break up the workflow into different files, like one for PR's, another for cutting tags, etc. I think it would be easier to read and maintain going forward, and it would help eliminate some of the `if ==` conditions in jobs which should only be executed on tag events. I think the AKS operator does this pattern well, https://github.com/rancher/aks-operator/tree/main/.github/workflows

Reply:
Good idea! Now the workflows are divided into different files based on the purpose. Please review the PR again.